engine/conditions.go

```diff
@@ -171,22 +171,19 @@ func (a AndCondition) Evaluate(ctx context.Context, log logr.Logger, condCtx Con
 			// TODO: determine if this is the right thing, I am assume the full rule should fail here
 			return ConditionResponse{}, fmt.Errorf("unable to find context value: %v", c.From)
 		}
-		response, err := c.ProviderSpecificConfig.Evaluate(ctx, log, condCtx)
+		response, err := c.Evaluate(ctx, log, condCtx)
 		if err != nil {
 			return ConditionResponse{}, err
 		}
+		// Must filter out based on the filepaths.
 		if c.As != "" {
 			condCtx.Template[c.As] = ChainTemplate{
 				Filepaths: incidentsToFilepaths(response.Incidents),
 				Extras:    response.TemplateContext,
 			}
 		}
 
-		matched := response.Matched
```

Comment:
I think I am not following why we remove 'not' here?

Reply:
Because I switched this to use https://github.com/konveyor/analyzer-lsp/pull/786/files#diff-bd3e00a860d81ad866acc697ae8784bd65ecd03dc56d5b4643b4768234bba2b8R174 which contains this logic.

Reply:
awesome, I understand now